### ISSUE TITLE
v055m: Zombie WireGuards in Simple mode

### ISSUE THREAD

**Issue by ignoramous** [celzero/rethink-app#1529]
> I have just realized that I can't delete WireGuard configs neither 1-by-1, nor by selecting all of them.
>
> Also, they got pretty crazy here — configs started to select and deselect on their own (without me clicking anything). Also, it writes status "ACTIVE", while RethinkDNS in not enabled and is paused.
> 
> https://github.com/celzero/rethink-app/assets/100644/8f6d396e-b82b-4bfc-9e7b-3b6d86dc6d35

_Originally posted by @soshial in https://github.com/celzero/rethink-app/issues/1524#issuecomment-2160042382_

**Comment by ignoramous** [comment]
> soshial, This issue seems concerning; Simple mode should not allow more than one WireGuard instance.
> 
> - What happens when you click on the name of the WireGuard in Simple mode?
> - Is this the behavior you have observed from the start?
> - Can you fetch the adb logcat after setting Rethink to Very Verbose mode in Configure -> Settings -> Log Level?

https://github.com/celzero/rethink-app/issues/1524#issuecomment-2160480272

**Comment by soshial** [comment]
I've got an idea that those are not really running WireGuards, but maybe a RecyclerView problem?
But I'll check the logs

**Comment by ignoramous** [comment]
Yeah, if you were always running in *Advanced* mode, then you should never be able to "view" *Simple* mode.

Likely the app crashed and then was left in this state when you re-opened it (you can the app *uptime* in the footer of the *bottomsheet* that comes up when you tap on the *downarrow* next to the *START* / *STOP* button) where you can see *Simple* mode but the conifgs are running in *Advanced* mode.

> But I'll check the logs

If you can capture logs starting from when you navigate to the screen and this happens, it'll help immensely. Make sure to put Rethink in *Verbose* mode from *Configure* -> *Settings* -> *Log level*.

**Comment by soshial** [comment]
Here are the logs. 
[rethinkdns.bugreport.zip](https://github.com/user-attachments/files/15787792/rethinkdns.bugreport.zip)

Also, in the Advanced regime, one WG was ON for some reason.

![photo_5422573632460283142_y](https://github.com/celzero/rethink-app/assets/100644/9ae74919-efa0-4348-84f1-9113bce70905)

![photo_5422573632460283140_y](https://github.com/celzero/rethink-app/assets/100644/d68d8cb3-f2b4-4da0-bf7b-2f4a6143a2a9)

**Comment by soshial** [comment]
> Yeah, if you were always running in Advanced mode, then you should never be able to "view" Simple mode.

I never ran in Advanced mode. Always in the Simple mode.

> Likely the app crashed and then was left in this state when you re-opened it.

Cannot check this anymore. When I saw this weird bug, I force-stopped the app myself to see whether this remove the issue, but it did not.

**Comment by hussainmohd-a** [comment]
Can you start the Rethink app and click on the Refresh icon on the Proxy screen to see if there are any changes? Also, please send us the bug report after the refresh is complete.

**Comment by soshial** [comment]
I clicked the refresh icon and now it doesn't allow me to select WireGuard configs anymore... Weird. 

![photo_5422573632460283186_y](https://github.com/celzero/rethink-app/assets/100644/c5b0788e-84c9-41a5-bba7-3204a3eecdf8)
[rethinkdns.bugreport(1).zip](https://github.com/user-attachments/files/15788306/rethinkdns.bugreport.1.zip)

**Comment by hussainmohd-a** [comment]
@soshial, are you now unable to enable the config when Rethink is active?

**Comment by soshial** [comment]
I was unable to select any config when Rethink was disabled (not active). It just showed this toast text when I clicked on the checkbox. But after some time it resolved by itself and I was able to open and activate configs.

You said that it's impossible to activate several configs in the "Simple regime", so only a single config possibel to activate — then I am wondering why do list elements have checkboxes instead of radio elements?

**Comment by ignoramous** [comment]
> You said that it's impossible to activate several configs in the "Simple regime", so only a single config possibel to activate — then I am wondering why do list elements have checkboxes instead of radio elements?

Mostly for the tick mark ✔️

**Comment by hussainmohd-a** [comment]
In v055o, made several improvements and fixes around WireGuard — this should resolve the issue. Closing this for now, but reopen if it pops up again.

**Comment by soshial** [comment]
But how can I even test it? There was no release for more than a year

**Comment by ignoramous** [comment]
You can build it from Hussain's tree (which is a few days old: https://github.com/hussainmohd-a/rethink-app/commits/v055n) or wait until Monday/Tuesday. These bugs are being close because `v055o` is ready to ship (code is mostly frozen).

**Comment by soshial** [comment]
I will wait out a couple of months first till it gets more stable.
